Utilities/Scripts/python_package_updater.py

```diff
@@ -44,6 +45,60 @@ def parse_pip_list_output(packages_to_update):
     return packages
 
 
+def external_project_filepaths(directory):
+    """Yield a generator of external project filepaths found in ``directory``.
+    """
+    for dirpath, _, filenames, in os.walk(directory):
+        for filename in filenames:
+            yield os.path.join(dirpath, filename)
+
+
+def hint_validation_mismatch_summary(mismatches):
+    """Format list of hint validation mismatches for display purpose.
+    """
+    return "\n".join([f"{filepath}\n  {first} != {second}" for filepath, first, second in mismatches])
+
+
+def validate_external_project_python_package_hints(directory):
+    """Check that external project files all have matching pair of hints.
+
+    Pair of hints are expected to be specified as CMake comment using the following
+    convention where ``<auto-generated-requirements>`` is replaced by the
+    ``python_package_updater`` script.
```

Comment:
I just realized this sentence may be ambiguous.

Reply:
```suggestion
    convention where ``<auto-generated-requirements>`` is replaced with updated contents by using the
    ``python_package_updater`` script.
```
You thinking something like the above?